src/huggingface_hub/hf_api.py

```diff
@@ -3433,6 +3434,125 @@ def hide_discussion_comment(
+            json={"key": key, "value": value},
+        )
+        hf_raise_for_status(r)
+
+    def delete_space_secret(
+        self, repo_id: str, key: str, *, token: Optional[str] = None
+    ) -> None:
+        """Deletes a secret from a Space.
+
+        Secrets allow to set secret keys or tokens to a Space without hardcoding them.
+        More for details, see https://huggingface.co/docs/hub/spaces-overview#managing-secrets.
+
+        Args:
+            repo_id (`str`):
+                ID of the repo to update. Example: `"bigcode/in-the-stack"`.
+            key (`str`):
+                Secret key. Example: `"GITHUB_API_KEY"`.
+            token (`str`, *optional*):
+                Hugging Face token. Will default to the locally saved token if not provided.
+        """
+        r = requests.delete(
+            f"{self.endpoint}/api/spaces/{repo_id}/secrets",
+            headers=self._build_hf_headers(token=token),
+            json={"key": key},
+        )
+        hf_raise_for_status(r)
+
+    def get_space_secrets(
+        self, repo_id: str, *, token: Optional[str] = None
+    ) -> Dict[str, str]:
+        """Gets all secrets from a Space.
+
+        Secrets allow to set secret keys or tokens to a Space without hardcoding them.
+        More for details, see https://huggingface.co/docs/hub/spaces-overview#managing-secrets.
+
+        Args:
+            repo_id (`str`):
+                ID of the repo to update. Example: `"bigcode/in-the-stack"`.
+            token (`str`, *optional*):
+                Hugging Face token. Will default to the locally saved token if not provided.
+        Returns:
+            `Dict[str, str]`: dictionary containing all key/value pairs.
+        """
+        r = requests.get(
+            f"{self.endpoint}/api/spaces/{repo_id}/secrets",
+            headers=self._build_hf_headers(token=token),
+        )
+        hf_raise_for_status(r)
+        return r.json()
+
+    def get_space_runtime(
+        self, repo_id: str, *, token: Optional[str] = None
+    ) -> SpaceRuntime:
+        """Gets runtime information about a Space.
+
+        Args:
+            repo_id (`str`):
+                ID of the repo to update. Example: `"bigcode/in-the-stack"`.
+            token (`str`, *optional*):
+                Hugging Face token. Will default to the locally saved token if
+                not provided.
+        Returns:
+            `SpaceRuntime`: dataclass containing runtime information about a Space
+             including Space stage and hardware.
+        """
+        r = requests.get(
+            f"{self.endpoint}/api/spaces/{repo_id}/runtime",
+            headers=self._build_hf_headers(token=token),
+        )
+        hf_raise_for_status(r)
+        data = r.json()
+        return SpaceRuntime(
+            stage=data["stage"],
+            hardware=data["hardware"]["current"],
+            requested_hardware=data["hardware"]["requested"],
+            raw=data,
+        )
```

Comment:
note that there's a PR open to also return this directly on the main `/spaces/xxx` info endpoint

Reply:
:+1: for using a single endpoint but I'd rather merge now and update in a follow-up PR once moon-landing PR is merged

Reply:
Ok with me as far as it's not released, just so we don't need to have backwards compatibility in two endpoints

Reply:
No, we were indeed talking about the same thing.

> I thought we would keep get_space_runtime but internally change which http endpoint it would call.

Yes exactly. My suggestion was to not do a **release** that uses the old HTTP endpoint. If we do a release, we risk users still using the old endpoint which means we force the server endpoint to have backwards compatibility. I would instead favor using the space info endpoint within `get_space_runtime` before next release.

Reply:
Ok that's clear, we are on the same page then ! 

[I created an issue](https://github.com/huggingface/huggingface_hub/issues/1252) and added it to 0.12 milestone in the meantime :slightly_smiling_face:

Reply:
note that the two endpoints will stay on the server anyways. i.E. i don't see a benefit of switching to the space info endpoint inside `get_space_runtime ` (if we do keep a dedicated `get_space_runtime ` method)

Reply:
OK, thanks for letting me know